### PR TITLE
issue 72

### LINKED ISSUE #72: Problem in toJSON and fromJSON with Model
in the **toJSON** function, in the case of is Model, the template will run isModel and isAnyType. so the value `json[r'{{baseName}}'] = {{{name}}};` can be found twice in the generated code.
Need to add ^isModel when running isAnyType. 

in case of **fromJSON**, in case of isModel, the generated code returns only the `json[r'{{{name}}}']` instead of `(json[r'{{{name}}}'] == null) ? null : {{{complexType}}}.fromJson(json[r'{{{name}}}']);`

plus, in the comparator == it is a good practice to add this.{{{name}}} before comparing

Happens only while running with `openapi-generator-cli-5.3.0.jar`

### PR DESCRIPTION
**Pull request by aviadkam**
add 'this.' to the equals operator.
fixing some issues with toJSON and fromJSON

fix #72

### CODE REVIEW

File: src/main/java/cd/connect/openapi/DartV3ApiGenerator.java [hunk 1]
```diff
@@ -637,6 +637,7 @@ public void postProcess() {
         .filter(Files::isRegularFile)
         .forEach(p -> {
           String modelFilename = p.toFile().getAbsolutePath().substring(stripLen);
+//          String modelFilename = p.getFileName().toString();
```
Comment (rvowles) [inline src/main/java/cd/connect/openapi/DartV3ApiGenerator.java]: please remove

**Comment by rvowles** [comment]
Could you please add an example in that breaks currently and is fixed by this? This seems very strange that something that isAnyType is matching a model, possibly a bug in the OpenAPI generator. You are using 5.2.1 yes?

**Comment by rvowles** [comment]
I just prefer the generator to get more resilient over time, thank you for your contribution :-)

**Comment by aviadkam** [comment]
Hi, 

Running with 5.2.1 seems to run OK.
I was running using 5.3

If we do not want to support 5.3 yet, then this PR can be closed

generated code using 5.2.1
`address =  (json[r'address'] == null) ? null : AddressViewModel.fromJson(json[r'address']);`

generated code using 5.3.0
`address = json[r'address'];`

**Comment by rvowles** [comment]
Ah, right - no I didn't even know 5.3 was around. I think it would probably be better to create a fix for this error in 5.3 rather than introduce something into the mustache that doesn't make sense (to me anyway). When walking through the parameters , we should check if isAny is selected as well as isModel, and if so, set isModel to false.

**Comment by aviadkam** [comment]
OK. I am closing this PR

**Pull request closed without merging**